src/sentry/demo/middleware.py

```diff
@@ -0,0 +1,30 @@
+from django.conf import settings
+
+from sentry.models import OrganizationMember
+from sentry.utils import auth
+
+
+class DemoMiddleware:
+    # automatically log in logged out users when they land
+    # on organization pages
+    def process_view(self, request, view_func, view_args, view_kwargs):
+        if not settings.DEMO_MODE:
+            return
+
+        # only handling org views
+        if "organization_slug" not in view_kwargs:
+            return
+
+        # if authed, no action required
+        if request.user.is_authenticated() and request.user.is_active:
```

Comment:
If the user is logged in but not part of the demo org would they hit errors viewing the demo org data? You might want to coerce their session into the demo org they are looking at.

Reply:
@markstory good point. It's sort of an edge case but it's possible someone could be logged in for on demo but receive a link for another org's demo. In that case, we should switch the user.